### DISCUSSION TITLE
Kan I center a component

### DISCUSSION THREAD

**Discussion by RuneClaeys** [histoire-dev/histoire#213] [Q&A]
I'm switching over from storybook and I'm looking for a way to copy this storybook behaviour to Histoire

```
export default {
  title: 'Button',
  component: Button,
  parameters: {
    layout: 'centered', // <-------------- how can i center the layout
  }

```
  
  
  Can this be done in -or added to histoire

**Comment by Akryum** [comment]
What would it look like?

**Reply by Akryum** [reply]
How would it work with the canvas resizing?

**Reply by RuneClaeys** [reply]
![image](https://user-images.githubusercontent.com/26223530/181753982-be8d5335-445b-4152-8ca4-99539bd770b5.png)
in storybook it just renders te component center of the canvas.

**Reply by RuneClaeys** [reply]
maybe you could just pass it to layout as an option?

`<Story :layout="{type: 'single', iframe: true, centered: true}"> ... </Story>`

**Reply by Akryum** [reply]
What about you center it with CSS in the story? Wouldn't that be enough?

**Reply by hrishikesh-k** [reply]
I had a similar requirement. I have a few buttons with shadows and something seems to be clipping all the content outside the box:

<img width="505" alt="image" src="https://user-images.githubusercontent.com/11403551/224129892-18a16fbb-e5f2-43fc-ade9-6a95d42d22a1.png">

I tried in `iframe` and `non-iframe` mode. Not sure which of the elements in hierarchy is setting `overflow:hidden`, I tried checking the DOM extensively.

**Reply by hrishikesh-k** [reply]
Okay, just found out, it's:

```
.__histoire-render-story:not(.__histoire-render-custom-controls)
```

that sets `overflow:auto`. Sadly overflow doesn't take into account box shadows. I might have to add a wrapper `div` with padding.
